several optimizations and some bugs fixed

Duke-Jones opened the pull request:
03.02.15: comment updated - see below

Hi, great tool !

I was only a little bit unsatisfied with the recognition rate
and so I testet different brightness and contrast settings
with the original TesseractOCR command line client.
After that I tested the different "traineddata" files as well.
It turned out that the "big.traineddata" from EliteOCR results in the best recognition rate.
I think it's very good now. Additionally I implemented a dynamic factor 
intstead a constant value for accepting the Leventhstein words.
So longer words allowing more incorrect characters. But so we can 
consider that they've also more correct charcters, too.
It works pretty good. 

While testing I was able to fix a bug and I' added a few other things.
I hope you like it.

Duke

PS: sorry for my bad english - I hope you can understand me ;-)
- button "Ignore As Trash" during ocr-correction added
- level for Levenshtein algorithm depends on the length of the word
- "really" empty strings will be ignored
- bug while procesing the results from EliteBrainerous removed
- optimized filter-processing for screenshots before ocr (tested on commandline tesseractOCR)
- using "big.traineddata" from EliteOCR, provides the best recognition (tested on commandline with tesseractOCR)
- now you can use enter-key for committing of commodities

03.02.15 - added something:

 1.8.4-0.0.1
- language selection (english-german-french)
- immediately autosave if data changed (prices, commodities, some settings)
- invoking crash bug fixed while OCR
- used traineddata-file for OCR is selectable
- EDDN reactivated if language is english
- fixed commodity base for better OCR (language dependent), possibility to add new commodities if confirmed
- bug fixed while ocr notification
- recognition of levels "high/medium/low" corrected (language dependent)
- behavior of "AutoUppercase"-Checkbox corrected
- takeover with ENTER if textfield has focus

Duke-Jones committed:
* optional "auto-uppercasing" checkbox during ocr-correction added
   * button "Ignore As Trash" during ocr-correction added
   * level for Levenshtein algorithm depends on the length of the word
   * "really" empty strings will be ignored
   * bug while procesing the results from EliteBrainerous removed
   * optimized filter-processing for screenshots before ocr (tested on commandline tesseractOCR)
   * using "big.traineddata" from EliteOCR, provides the best recognition (tested on commandline with tesseractOCR)

Duke-Jones committed:
* consider that the textfield support auto-uppercase for state detection
   * double uppercase eliminated

Duke-Jones committed:
1.8.4-0.0.1
   * language selection (english-german-french)
   * immediately autosave if data changed (prices, commodities, some settings)
   * invoking crash bug fixed while OCR
   * used traineddata-file for OCR is selectable
   * EDDN reactivated if language is english
   * fixed commodity base for better OCR (language dependent), possibility to add new commodities if confirmed
   * bug fixed while ocr notification
   * recognition of levels "high/medium/low" corrected (language dependent)
   * behavior of "AutoUppercase"-Checkbox corrected
   * takeover with ENTER if textfield has focus

Duke-Jones committed:
1.8.4-0.0.2
   * added "Microbial Furnaces", "Advanced Catalysers" and "Resonating Separators" as commodity
   * added missing Dataset "dsCommodities" and class "enumBindTo" to repositiory (relevant for compiling source)

Duke-Jones committed:
v1.84-0.03
   * loading display- and app-data from ED: improved handling of errors while re-reading files
   * check both versions ("official" and "international") of RegulatedNoise for updates
   * New datafiles will not be activated until savings done without errors.
     old data files (AutoSave.csv/CommandersLogAutoSave.xml, RegulatedNoiseSettings.xml) 
     will be saved as "*_bak.<ext>" instead of deleted
   * fixed : commodity data was not written if "include extended info" was disabled
   * changed Comboboxes from DropDown to DropDownList with "suggest" AutoCompleteMode
   * comboboxes that wasn't sorted can be sorted (Settings)
   * added new error handling in some functions
   * automatically generated "Jumped To ..." event is working again
   * crash while re-reading GameSettings.xml fixed

Donavan commented:
Pulling this puts the project into an uncompilable state unless you define DukeJones.

Duke-Jones committed:
* fixed a compiling error if constant "DukeJones" wasn't defined

Duke-Jones commented:
Oh, I'm sorry. 
I fixed this.

Duke-Jones committed:
1.84-0.04
   * in the distance dropdown can own distances typed in again (issue #9)
   * included current station and system data (17.02.15)
   * included a performancetimer for checking for bottlenecks (not active now)
   * saving the sort order of "Cmdr's Log" on exit 
   * last written "Cmdr's Log" event is the active event 
   * autogenerated "Jumped To" event can be disabled

Duke-Jones committed:
v1.84-0.05

1.84-0.05
* set the correct path for update for DJ version
* speed of OCR process improved

Duke-Jones committed:
v1.84_0.06

v1.84_0.06
* fixed a crash reason while running improved OCR process
* included current station and system data (22.02.15)

Duke-Jones committed:
1.84_0.07_1

* find the path to the ED game directoy (first start) even if it's no
standard path
* Pixeltest is now sure visible in foreground of RN
* saving and restoring position and state of main window

Duke-Jones committed:
added forgotten file

Duke-Jones committed:
1.84_0.07_2
   * added support for EDDB data (systems/stations/commodities)
   * implemented a new internal structure for managing the system and station data

Duke-Jones committed:
1.84_0.07_3
   * fixed a error in the new Milkyway-class
   * locations of the systems are saved in a quick cache

Duke-Jones committed:
1.84_0.07_4 (prerelease)
   * comboboxes (station/system) can be sorted by 
      - station only 
	  - system and station
	  - distance
	  - additional can "n" last visited stations shown on top of the list
   * internal structure is completed for considering the "size of landingpads",
     "distances of station to stars" and other data based on EDDB
   * more settings are saved when closing the application

Duke-Jones committed:
1.84_0.08
   * Comboboxes hold their selected item if reordering or anything other
     change the list of the DropDown items
   * "Edit Row" for commodities is working again
   * CSV-Import: if there already recordset that is younger than 
     the recordset to import so the older recordset gets ignored
   * if there are more than one recordset for a commodity on a station
     so only the younger one gets considered for route calculations
   * order of colums for station and system names in dropdown 
     depends on the order of the dropdown itself 
   * setting of checkbox "perLightYearRoundTrip" is also saved
   * repeatedly language switching result not anymore in a crash
   * a different language than englisch is saved again

+ the changes from "1.84_0.07_1" to "1.84_0.07_4"

Duke-Jones committed:
1.84_0.09_1
   * fixed: "perLightYearRoundTrip" was not switchable anymore
   * improved segmenting capability of EliteBrainerous
   * splittet OCR base picture into 2 different optimized pictures 
     in cause of different working methods of EliteBrainerous and TesseractOCR
   * added the possibility to start one-time-operations for necessary
     setting changes for new versions

Duke-Jones committed:
1.84_0.09_2
   * improved segmenting capability of EliteBrainerous again
   * splitting for OCR reasons from 1.84_0.09_1 removed again
   * added a test for checking the color-cutoff value with samples
   * clean up some code

Duke-Jones committed:
added some forgotten files for 1.84_0.09_2

Duke-Jones committed:
* remove compiling issues for 1.84_0.09_2

Duke-Jones committed:
1.84_0.09
   * added up-to-date data files from EDDB (v3) and expand internal structure 
   * changed default "dark pixel amount" to 22
   * changed default "dark pixel threshold" to 0.6
   * changed from "1.84_0.09_1" to "1.84_0.09_2"

Duke-Jones committed:
1.84_0.10_1
   * compiled nn_traing.exe with option "noconsole" 
   * added classes for calculating plausibility limits for the prices

Duke-Jones committed:
1.84_0.10_2
   * checking plausibility for the prices          ** "Auto Import" will be possible in future ** 
   * mark implausible prices in "Result Edit Window"
   * removed "Auto Uppercase" option - it's now always on

Duke-Jones committed:
1.84_0.10_3
   * "checking for a one" cleans now up the GUI
   * fixed a few errors around the plausibility check
   * single result lines of ocr are now erasable
   * added function to auto-activate OCR tab if recognition starts
   * already recognized commodities will not added again anymore during a single ocr cycle
   * added a form for viewing and editing the price warning levels

Duke-Jones committed:
1.84_0.10_4
   * added up-to-date data files from EDDB
   * additional table view for viewing and editing the plausibility limits
   * added a abstract window base class for supporting default actions like saving window positions
   * timer for checking the OCRBuffer now is stopping again if nothing more to do
   * OCR-tab is automatically shown if ocr starts (optional)

Duke-Jones committed:
1.84_0.10
   * completed base class for easy position saving
   * saving now the positions of all windows
   * added a donation button 
   * included a new class for saving dictionaries as XML-file

Duke-Jones committed:
1.84_0.11_1
   * fixed: removing price data crashing the program
   * removing and changing of price data showing changes immediately
   * a few gui cleanups and small repositionings
   * removed "commodities_RN.json" (it' auto generated at the first start or it exists already)

Duke-Jones committed:
1.84_0.11
   * if a station is renamed the new name gets now the correct format
   * fixed another crash reason while changing data
   * another few gui cleanups and object reconnectings

Duke-Jones committed:
1.84_0.12_1
   * getting current location in the current system from logfile, too
   * clock added (local and ingame time)

Duke-Jones committed:
1.84_0.12_2
   * delete setting.ini if calibrating new (previous calculated digit width can result in bad OCR)
   * commodity warnlevels are now saved if changed
   * fixed crash reason if deleting all commodities from a station
   * Cmdr's Log now showing dates in local culture settings

Duke-Jones committed:
1.84_0.13_1
   * fixed some issues while looking for location in logfiles
   * changed GUI from CorrectingOCRResults in cause of viewing issues

Duke-Jones committed:
1.84_0.13_2
   * fixed some issues while looking for location in logfiles

Duke-Jones committed:
1.84_0.13_2
   * first rows for "station distance to star" filter
   * saving column widths of Cmdr's Log if changed
   * optimized standard column widths of Cmdr's Log
   * sorting of date columns now should be correct

Duke-Jones committed:
1.84_0.13_4
   * added first version of gui for showing and editing system data
   * "station distance to star" filter full included 
     ! some stations have still no distance information yet (station-edit needed -> next versions). 
	   On settings tab you can choose if they should be incuded or excluded
   * still changings on log scanning issue (maybe finished now)

Duke-Jones committed:
1.84_0.13
   * fixed: sometimes OCR calibration tab is opened instead of "capture and correct" tab
   * fixed: crash if asking for unknown systemdata
   * fixed: auto adding Cmdr's Log after jump scrolls the whole list
   * added current EDDN json files

Duke-Jones committed:
1.84_0.14_1
   * changes on log scanning issue + additional log info

Duke-Jones committed:
1.84_0.14_2
   * added a progress view for calculating best routes with cancel option
   * filter added : max. distance for trade routes

Duke-Jones committed:
* added missing ProgressBar

Duke-Jones committed:
1.84_0.15_1
   * auto show systemdata
   * port of webserver is adjustable
   * show number of stations in system
   * completed splash screen
   * optimized program start behaviour

DrAvEn83 commented:
hi Duke Nice job

but can you add a butten for Delet 1 Commodities,
and LS Filter add 

im happy too see you work good

Duke-Jones committed:
1.84_0.15_2
   * sorting of comboboxes is saved now
   * fixed a unwanted label on splash

Duke-Jones committed:
1.84_0.16
   * translating of EDDN messages, EDDN now possible with other languages than english
   * fixed a bug in the internal translating logic
   * added a plausibility filter to the incoming EDDN logic
   * added last EDDB data files

Duke-Jones committed:
1.84_0.17_1
   * set "unset"-value for EDDB station and system properties to null
   * option to cancel while asked for needed directorys on first start
   * systems in general now viewable, not only the current system
   * cleaned up system tab
   * added new helper classes "controls_ext", "Extensions", "ObjectCompare"

Duke-Jones committed:
1.84_0.17_2
   * data system can be added and changed - station data still open

Duke-Jones committed:
1.84_0.17_3
   * many steps for adding and changing station data are done

Duke-Jones committed:
1.84_0.17_4
   * it's possible to change data of existing stations

Duke-Jones committed:
1.84_0.17_5
   * it's possible to add new stations
   * default value for EDDN is no longer <test> because it's only for dev testing
   * set value for EDDN schema from <test> to <normal> and add a hint for users who have EDDN active

Duke-Jones committed:
1.84_0.18_1
   * positons of Messageboxes and Inputboxes are centered over the parent window
   * added "Alt-O" shortcut for "Continue" on OCR tab
   * get stationname for OCR from "location from logfile" if the stationname is 
     confirmed as "existing in system" by database
   * added possibility to purge old data on settings tab
   * added "None" to items in station and system data
   * correction on <test> EDDN schema check

Commit:
fix invalid cast from EDCommoditiesExt to int

Commit:
Merge branch 'master' of https://github.com/zericco/RegulatedNoise into commodities_view_fix

Duke-Jones committed:
1.84_0.18_2
   * put eddn sending routine into a backgroundthread
   * if test-schema is not active, so all data coming from test schema will also be ignored
   * Commander's Log sets station information on new entrys
   * only known stations in a system are suggested in "Commander's Log" station-dropdown
   * new AutoEvents: 
            - "Visited" (Stations)
			- "Market Data Collected"
   * optimized refresh behaviour of Commander's Log
   * added some RN specific infos on the start tab
   * added latest EDDB data files

Duke-Jones committed:
added newest eddb data
and set the correct version

Duke-Jones committed:
added newest eddb data (2)

Duke-Jones committed:
added newest eddb data (3)

Duke-Jones committed:
Merge pull request #30 from zericco/master

fix invalid cast from EDCommoditiesExt to int
